nwb_conversion_tools/datainterfaces/ecephys/basesortingextractorinterface.py

```diff
@@ -27,16 +26,38 @@ def __init__(self, **source_data):
 
     def get_metadata_schema(self):
         """Compile metadata schema for the RecordingExtractor."""
-        metadata_schema = get_base_schema(
-            properties=dict(SpikeEventSeries=get_schema_from_hdmf_class(SpikeEventSeries))
+        metadata_schema = super().get_metadata_schema()
+
+        # Initiate Ecephys metadata
+        metadata_schema["properties"]["Ecephys"] = get_base_schema(tag="Ecephys")
+        metadata_schema["properties"]["Ecephys"]["required"] = []
+        metadata_schema["properties"]["Ecephys"]["properties"] = dict(
+            UnitProperties=dict(
+                type="array",
+                minItems=0,
+                renderForm=False,
+                items={"$ref": "#/properties/Ecephys/properties/definitions/UnitProperties"},
+            ),
+        )
+        # Schema definition for arrays
+        metadata_schema["properties"]["Ecephys"]["properties"]["definitions"] = dict(
+            UnitProperties=dict(
+                type="object",
+                additionalProperties=False,
+                required=["name"],
+                properties=dict(
+                    name=dict(type="string", description="name of this units column"),
+                    description=dict(type="string", description="description of this units column"),
+                ),
+            ),
```

Comment:
shouldn't this be an array?

Reply:
That would be a question for @luiztauffer 

I basically copied the way the electrode custom columns are specified by the schema seen [here](https://github.com/catalystneuro/nwb-conversion-tools/blob/fix_sorting_interface/nwb_conversion_tools/datainterfaces/ecephys/baserecordingextractorinterface.py#L46-L66)

Reply:
As I understand though, this level of "object" specifies that what is inside the 'UnitProperties' array are dictionaries with keys 'name' and 'description', with the code above this specifying that the 'UnitProperties' field itself is an array